airbyte-integrations/connectors/source-shopify/source_shopify/shopify_graphql/bulk/record.py

```diff
@@ -0,0 +1,93 @@
+#
+# Copyright (c) 2023 Airbyte, Inc., all rights reserved.
+#
+
+
+import re
+from inspect import isgeneratorfunction
+from json import loads
+from typing import Any, Callable, Iterable, Mapping, Union
+
+from .query import PARERNT_KEY
+from .tools import BulkTools
+
+
+class ShopifyBulkRecord:
+    def __init__(self) -> None:
+        self.tools: BulkTools = BulkTools()
+
+    @staticmethod
+    def record_resolve_id(record: Mapping[str, Any]) -> Mapping[str, Any]:
+        """
+        The ids are fetched in the format of: " gid://shopify/Order/<Id> "
+        Input:
+            { "Id": "gid://shopify/Order/19435458986123"}
+        We need to extract the actual id from the string instead.
+        Output:
+            { "id": 19435458986123, "admin_graphql_api_id": "gid://shopify/Order/19435458986123"}
+        """
+        # save the actual api id to the `admin_graphql_api_id`
+        record["admin_graphql_api_id"] = record["id"]
+        # extracting the int(id) and reassign
+        record["id"] = int(re.search(r"\d+", record.get("id")).group())
+        return record
+
+    @staticmethod
+    def resolve_substream(record: Mapping[str, Any], record_identifier: str = None) -> Mapping[str, Any]:
+        # return records based on `record_identifier` filter.
+        if record_identifier:
+            return record if record_identifier in record.get("admin_graphql_api_id") else None
```

Comment:
can you add a comment on `admin_graphql_api_id`? it's not clear what the field means from looking at the code

Reply:
Added.